test/e2e/validators.go

```diff
@@ -129,6 +131,20 @@ func createClusterBuildStrategy(
 	}
 }
 
+func amendBuildRun(br *operator.BuildRun, imageRepoBuildRun string)  {
```

Comment:
I understand why you do this function, but it seems to me that is too expensive. You kind of "reproduce"  the code you put in `generate_taskrun.go ` one more time in `validators.go`. We have a similar approach for the `spec.output.ImageURL` for the Build in e2e tests, but it make sense while we should be able to override the YAMLs sample files with an endpoint where we have access(e.g. ibm container registry). For the `spec.output.ImageURL` I said is expensive because:
- you introduce a new ENV variable in e2e tests
- you introduce new functions in `validators.go` for dealing with the above ENV variable
- you introduce a new e2e test case

I think this could have been easily validated by:
- adding a unit-test in `generate_taskrun_test.go` (which you did, which is enough for testing)

Reply:
Actually, I agree with you. Because look like I just check the buildRun is successful or not. I don't check if the ImageURL in taskRun is the imageURL from buildRun instead of build. And this verification has been done in unit test.
So now, I remove the e2e test for it.